### ISSUE TITLE
Not able to replicate the HRFuser results on DENSE

### ISSUE THREAD

**Issue by kvnptl** [timbroed/HRFuser#4]
Hello,

First of all, thank you for open-sourcing the code and providing an easy-to-follow README for testing the HRFuser model.

### Issue Summary:

On DENSE, the results are good. However, it still does not match the published article in the paper.

### Environment:
- Multi-GPU: 4x NVIDIA V100
- Python 3.8
- pytorch 1.10.0
- torchvision 0.11.0
- cuda 10.2
- gcc 10.1.0
- Other hyperparameters and model settings (as per the code repository)

### Steps to Reproduce:

1. Cloned the repository.
2. Installed all dependencies.
3. Ran the training script with 60 epochs.

### Overall Results:

- **Model: [Tiny]**
 
I tried 60 epochs with batch size 8, as mentioned in the paper, with the same settings as given in the code with multi-GPU. 

| Clear       |            |            | Light Fog  |            |            | Dense Fog  |            |            | Snow/Rain Fog |            |            |
|-------------|------------|------------|------------|------------|------------|------------|------------|------------|--------------|------------|------------|
| Easy        | Moderate   | Hard       | Easy       | Moderate   | Hard       | Easy       | Moderate   | Hard       | Easy         | Moderate   | Hard       |
| 82.82       | 79.81      | 75.88      | 86.15      | 84.58      | 81.50      | 74.30      | 68.67      | 67.23      | 83.61        | 78.40      | 75.07      |

###   With given pre-trained weights,

| Clear       |            |            | Light Fog  |            |            | Dense Fog  |            |            | Snow/Rain Fog |            |            |
|--------|------|-------|------|--------|--------|------------|------------|------------|--------------|------------|------------|
| Easy   | Moderate  | Hard   | Easy | Moderate | Hard   | Easy | Moderate   | Hard       | Easy         | Moderate   | Hard       |
| 82.88 | 79.58  | 74.83      | 86.48      | 83.46      | 81.20      | 75.47      | 69.75      | 68.45      | 84.55        | 78.76      | 75.34      |

For DENSE, training from scratch and given pre-trained weights results are matching, but these numbers are off from the numbers published in the paper. These results are behind the compared method `Deep Entropy Fusion` in the paper.

Please let me know if any other info would be helpful for debugging.

Any help in debugging the issue would be greatly appreciated.

Thank you!

**Comment by tbroed** [comment]
Hello,

thanks for your message.  

Please try the training with a batch size of 12 (4 GPU's and 3 samples_per_gpu) as it is also mentioned in the paper. If you adjust the batch size, please also adjust the learning rate accordingly.

Regarding the use of the pre-trained model on DENSE, where you say that you did not get the exact results from the paper: I got the exact results mentioned in the paper when I evaluated the provided checkpoint, so maybe something went wrong with your dataset preparation?

Best,
Tim

**Comment by kvnptl** [comment]
Hello,

Thank you for the feedback. 

Regarding the use of the pre-trained model on DENSE, how do I make sure that my dataset preparation is correct? Is there any way to debug or confirm the dataset?

I followed the readme instructions to create the dataset.

DENSE dataset number of samples,
- `cam_stereo_left_lut` - 12997
- `lidar_samples` - 12994
- `radar_samples` - 12994

Here is an example sample: `2018-02-04_11-29-58_00000`,
- Camera
![2018-02-04_11-29-58_00000_cam](https://github.com/timbroed/HRFuser/assets/47410011/7275c948-47dc-4fa0-a45c-ebc4165ff135)

- LIDAR image
![2018-02-04_11-29-58_00000_lidar](https://github.com/timbroed/HRFuser/assets/47410011/7f5afd9e-1677-47b9-bb22-f4f9cff57841)


- RADAR image
![2018-02-04_11-29-58_00000_radar](https://github.com/timbroed/HRFuser/assets/47410011/6152b4a2-954f-4c8a-9286-00e80987c8e3)

Is there any problem with the LIDAR and RADAR images?

**Comment by kvnptl** [comment]
Hello,

I am also getting the same result.

I just realized that in the paper, it's mentioned that the evaluation results are only for the `car` class.  

> We train on the common KITTI classes car, pedestrian, and cyclist, and evaluate only on car using the KITTI evaluation framework [16], similar to [1].

But I was considering the overall score with all 3 classes (car, pedestrian, and bicycle). 

Sorry for the confusion.

Thanks,
K
